regression/pages/common/utils.py

```diff
@@ -0,0 +1,496 @@
+
+    def delete_coupon(self, coupon_id):
+        """
+        Delete the coupon created in the tests and all associated stock
+        records and vouchers.
+        Args:
+            coupon_id:
+        """
+        self.ecommerce_api_client.coupons(coupon_id).delete()
+
+    def get_coupon_codes(self, coupon_id):
+        """
+        Get coupon codes from coupon report
+        Args:
+            coupon_id:
+        """
+        coupon_codes = []
+        items = (
+            self.get_coupon_report(coupon_id)['attribute_values'][0]['value'])
+        for item in items:
+            coupon_codes.append(item['code'])
+        return coupon_codes
+
+
+def yesterday_date():
+    """
+    Get and return yesterday's date in dd-mmm-yyyy format
+    """
+    return (datetime.date.today() - datetime.timedelta(1)).strftime("%d-%b-%Y")
+
+
+def get_required_cookies(cookies_dict_list):
+    """
+    Format required cookies
+    Args:
+        cookies_dict_list:
+    """
+    response_dict = {}
+    for cookie in cookies_dict_list:
+        if cookie['name'] == 'ecommerce_csrftoken':
+            response_dict.update({'ecommerce_csrftoken': cookie['value']})
+        elif cookie['name'] == 'ecommerce_sessionid':
+            response_dict.update({'ecommerce_sessionid': cookie['value']})
+    required_cookies = '; '.join('{}={}'.format(
+        key, val) for key, val in response_dict.items())
+    return required_cookies
+
+
+def get_coupon_request(target_url, cookie_str):
+    """
+    Send a Get request and return the response if successful
+    Args:
+      target_url:
+      cookie_str:
+    """
+    kwargs = {}
+    headers = {
+        'Accept': 'text/html,application/xhtml+xml,'
+                  'application/xml;q=0.9,image/webp,*/*;q=0.8',
+        'Accept-Language': 'en-US,en;q=0.8',
+        'Accept-Encoding': 'gzip, deflate, sdch, br',
+        'Cookie': cookie_str
+    }
+    kwargs.update({
+        'headers': headers,
+        'auth': HTTPBasicAuth(AUTH_USERNAME, AUTH_PASSWORD),
+    })
+    response = requests.get(target_url, **kwargs)
+    if response.ok:
+        return response.content
+    else:
+        msg = "Could not use restful API.  Status code: {0}".format(
+            response.status_code)
+        raise ApiException(msg)
+
+
+def get_enrollment_codes_from_email(coupon_string):
+    """
+    Read enrollment codes and urls from email
+    Ignore first four lines of file as these are just headers
+    Args:
+        coupon_string:
+    Returns:
+        coupon codes and urls:
+    """
+    coupons = {}
+    for i, row in enumerate(coupon_string.splitlines()):
+        # Read the coupons starting from line 5
+        if i > 3:
+            if row:
+                new_row = row.split(',')
+                coupons.update({new_row[0]: new_row[1]})
+    if coupons:
+        return coupons
+    else:
+        return 'Coupons not found'
+
+
+def skip_cleanup_if_test_passed():
```

Comment:
Cleanup should be idempotent. You should be able to run it whether the test passed or failed. Its job is to ensure a given end-state.